In [1]:
#Imports

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

## Data Exploratory Analysis

In [2]:
df=pd.read_csv("C:/Users/jwsam/Desktop/Ecom/kaggle_datasets/olist_order_reviews_dataset.csv")

In [3]:
df_order_reviews=pd.read_csv("C:/Users/jwsam/Desktop/Ecom/kaggle_datasets/olist_order_reviews_dataset.csv")

In [4]:
df_order_reviews.head(70)

review_id                          order_id  \
0   7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1   80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2   228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3   e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4   f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   
..                               ...                               ...   
65  9924ac6169f2edc95ca3394e529f9580  a3ba7fa5266d62188949da3d6ccc3884   
66  e12a8d938f3092e010627d001f91d137  b3dca2e199b13ddabab4ed77d79ee9f6   
67  109b5ce2dd11bb8460eff3b86da6fefc  25362fbf6aac4b01a28dee1e076acc26   
68  6d06808638ec0701bccd70bc8d462c28  97d2f8fe76f2f253b8291e17b5383884   
69  750d4311e8ddb6cba549ee86f06f0cca  4b32a4cdfc1b80066a8ee3a4b7a2094f   

    review_score review_comment_title  \
0              4                  NaN   
1              5                  NaN   
2              5                  NaN   
3              5                  NaN   
4              5                  NaN   
..           ...                  ...   
65             5                  NaN   
66             5                  NaN   
67             5      Ótimo Produto!    
68             1                  NaN   
69             5                  NaN   

                               review_comment_message review_creation_date  \
0                                                 NaN  2018-01-18 00:00:00   
1                                                 NaN  2018-03-10 00:00:00   
2                                                 NaN  2018-02-17 00:00:00   
3               Recebi bem antes do prazo estipulado.  2017-04-21 00:00:00   
4   Parabéns lojas lannister adorei comprar pela I...  2018-03-01 00:00:00   
..                                                ...                  ...   
65                                                NaN  2018-02-27 00:00:00   
66                                                NaN  2018-06-03 00:00:00   
67                           Recomendo o vendedor...   2018-08-17 00:00:00   
68  O produto não chegou no prazo estipulado e cau...  2016-12-09 00:00:00   
69  Produto entregue como solicitado, e com muita ...  2017-08-09 00:00:00   

   review_answer_timestamp  
0      2018-01-18 21:46:59  
1      2018-03-11 03:05:13  
2      2018-02-18 14:36:24  
3      2017-04-21 22:02:06  
4      2018-03-02 10:26:53  
..                     ...  
65     2018-03-02 08:33:35  
66     2018-06-06 12:41:24  
67     2018-08-17 21:47:08  
68     2016-12-19 14:10:43  
69     2017-08-09 12:03:45  

[70 rows x 7 columns]

In [5]:
df_order_reviews.info()

<class 'pandas.DataFrame'>
RangeIndex: 99224 entries, 0 to 99223
Data columns (total 7 columns):
 #   Column                   Non-Null Count  Dtype
---  ------                   --------------  -----
 0   review_id                99224 non-null  str  
 1   order_id                 99224 non-null  str  
 2   review_score             99224 non-null  int64
 3   review_comment_title     11568 non-null  str  
 4   review_comment_message   40977 non-null  str  
 5   review_creation_date     99224 non-null  str  
 6   review_answer_timestamp  99224 non-null  str  
dtypes: int64(1), str(6)
memory usage: 5.3 MB


For the review_id, order_id, review_score, review_creation_date, review_answer_timestamp the data is nont null with equal value of 99223 instances
While the review_comment_title and the review_comment_message have null values

In [6]:
df_order_reviews.describe()

review_score
count  99224.000000
mean       4.086421
std        1.347579
min        1.000000
25%        4.000000
50%        5.000000
75%        5.000000
max        5.000000

In [7]:
df_order_reviews['review_comment_title'].isna().sum()

np.int64(87656)

In [8]:
unique_values = df['review_comment_title'].dropna().unique()

In [9]:
unique_values.unique()

<StringArray>
[                'recomendo',           'Super recomendo',
   'Não chegou meu produto ',                     'Ótimo',
                'Muito bom.',                       'Bom',
              'Maravilhoso!',          'Entrega perfeita',
                  'MT lindo',           'Ótimo Produto! ',
 ...
 'Não posso recomendar aind',      'Rápido porém não bom',
           'entregue rapido',                        '7 ',
      'Entrega super Rápida',              'Nota máxima!',
        'muito bom produto ', 'Não foi entregue o pedido',
           'OTIMA EMBALAGEM',            'Foto enganosa ']
Length: 4527, dtype: str

In [10]:
df_order_reviews['review_comment_title'] = df_order_reviews['review_comment_title'].fillna("no title")

In [11]:
df_order_reviews

review_id                          order_id  \
0      7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1      80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2      228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3      e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4      f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   
...                                 ...                               ...   
99219  574ed12dd733e5fa530cfd4bbf39d7c9  2a8c23fee101d4d5662fa670396eb8da   
99220  f3897127253a9592a73be9bdfdf4ed7a  22ec9f0669f784db00fa86d035cf8602   
99221  b3de70c89b1510c4cd3d0649fd302472  55d4004744368f5571d1f590031933e4   
99222  1adeb9d84d72fe4e337617733eb85149  7725825d039fc1f0ceb7635e3f7d9206   
99223  efe49f1d6f951dd88b51e6ccd4cc548f  90531360ecb1eec2a1fbb265a0db0508   

       review_score review_comment_title  \
0                 4             no title   
1                 5             no title   
2                 5             no title   
3                 5             no title   
4                 5             no title   
...             ...                  ...   
99219             5             no title   
99220             5             no title   
99221             5             no title   
99222             4             no title   
99223             1             no title   

                                  review_comment_message review_creation_date  \
0                                                    NaN  2018-01-18 00:00:00   
1                                                    NaN  2018-03-10 00:00:00   
2                                                    NaN  2018-02-17 00:00:00   
3                  Recebi bem antes do prazo estipulado.  2017-04-21 00:00:00   
4      Parabéns lojas lannister adorei comprar pela I...  2018-03-01 00:00:00   
...                                                  ...                  ...   
99219                                                NaN  2018-07-07 00:00:00   
99220                                                NaN  2017-12-09 00:00:00   
99221  Excelente mochila, entrega super rápida. Super...  2018-03-22 00:00:00   
99222                                                NaN  2018-07-01 00:00:00   
99223  meu produto chegou e ja tenho que devolver, po...  2017-07-03 00:00:00   

      review_answer_timestamp  
0         2018-01-18 21:46:59  
1         2018-03-11 03:05:13  
2         2018-02-18 14:36:24  
3         2017-04-21 22:02:06  
4         2018-03-02 10:26:53  
...                       ...  
99219     2018-07-14 17:18:30  
99220     2017-12-11 20:06:42  
99221     2018-03-23 09:10:43  
99222     2018-07-02 12:59:13  
99223     2017-07-03 21:01:49  

[99224 rows x 7 columns]

In [12]:
df_order_reviews['review_comment_message'] = df_order_reviews['review_comment_message'].fillna("no message")

In [13]:
df_order_reviews

review_id                          order_id  \
0      7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1      80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2      228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3      e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4      f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   
...                                 ...                               ...   
99219  574ed12dd733e5fa530cfd4bbf39d7c9  2a8c23fee101d4d5662fa670396eb8da   
99220  f3897127253a9592a73be9bdfdf4ed7a  22ec9f0669f784db00fa86d035cf8602   
99221  b3de70c89b1510c4cd3d0649fd302472  55d4004744368f5571d1f590031933e4   
99222  1adeb9d84d72fe4e337617733eb85149  7725825d039fc1f0ceb7635e3f7d9206   
99223  efe49f1d6f951dd88b51e6ccd4cc548f  90531360ecb1eec2a1fbb265a0db0508   

       review_score review_comment_title  \
0                 4             no title   
1                 5             no title   
2                 5             no title   
3                 5             no title   
4                 5             no title   
...             ...                  ...   
99219             5             no title   
99220             5             no title   
99221             5             no title   
99222             4             no title   
99223             1             no title   

                                  review_comment_message review_creation_date  \
0                                             no message  2018-01-18 00:00:00   
1                                             no message  2018-03-10 00:00:00   
2                                             no message  2018-02-17 00:00:00   
3                  Recebi bem antes do prazo estipulado.  2017-04-21 00:00:00   
4      Parabéns lojas lannister adorei comprar pela I...  2018-03-01 00:00:00   
...                                                  ...                  ...   
99219                                         no message  2018-07-07 00:00:00   
99220                                         no message  2017-12-09 00:00:00   
99221  Excelente mochila, entrega super rápida. Super...  2018-03-22 00:00:00   
99222                                         no message  2018-07-01 00:00:00   
99223  meu produto chegou e ja tenho que devolver, po...  2017-07-03 00:00:00   

      review_answer_timestamp  
0         2018-01-18 21:46:59  
1         2018-03-11 03:05:13  
2         2018-02-18 14:36:24  
3         2017-04-21 22:02:06  
4         2018-03-02 10:26:53  
...                       ...  
99219     2018-07-14 17:18:30  
99220     2017-12-11 20:06:42  
99221     2018-03-23 09:10:43  
99222     2018-07-02 12:59:13  
99223     2017-07-03 21:01:49  

[99224 rows x 7 columns]

### Due to the fact that the review creation date and review answer timestamp have no relevance to what the consumers felt about the shopping experience

In [14]:
df_order_reviews.drop(columns=['review_creation_date','review_answer_timestamp'])

review_id                          order_id  \
0      7bc2406110b926393aa56f80a40eba40  73fc7af87114b39712e6da79b0a377eb   
1      80e641a11e56f04c1ad469d5645fdfde  a548910a1c6147796b98fdf73dbeba33   
2      228ce5500dc1d8e020d8d1322874b6f0  f9e4b658b201a9f2ecdecbb34bed034b   
3      e64fb393e7b32834bb789ff8bb30750e  658677c97b385a9be170737859d3511b   
4      f7c4243c7fe1938f181bec41a392bdeb  8e6bfb81e283fa7e4f11123a3fb894f1   
...                                 ...                               ...   
99219  574ed12dd733e5fa530cfd4bbf39d7c9  2a8c23fee101d4d5662fa670396eb8da   
99220  f3897127253a9592a73be9bdfdf4ed7a  22ec9f0669f784db00fa86d035cf8602   
99221  b3de70c89b1510c4cd3d0649fd302472  55d4004744368f5571d1f590031933e4   
99222  1adeb9d84d72fe4e337617733eb85149  7725825d039fc1f0ceb7635e3f7d9206   
99223  efe49f1d6f951dd88b51e6ccd4cc548f  90531360ecb1eec2a1fbb265a0db0508   

       review_score review_comment_title  \
0                 4             no title   
1                 5             no title   
2                 5             no title   
3                 5             no title   
4                 5             no title   
...             ...                  ...   
99219             5             no title   
99220             5             no title   
99221             5             no title   
99222             4             no title   
99223             1             no title   

                                  review_comment_message  
0                                             no message  
1                                             no message  
2                                             no message  
3                  Recebi bem antes do prazo estipulado.  
4      Parabéns lojas lannister adorei comprar pela I...  
...                                                  ...  
99219                                         no message  
99220                                         no message  
99221  Excelente mochila, entrega super rápida. Super...  
99222                                         no message  
99223  meu produto chegou e ja tenho que devolver, po...  

[99224 rows x 5 columns]

In [15]:
df_order_reviews.to_csv("df_order_reviews_cleaned.csv", index=False)